### DISCUSSION TITLE
Web browser / client-side: How to check if a session cookie already exists

### DISCUSSION THREAD

**Discussion by caniko** [fastapi-users/fastapi-users#1060] [Q&A] [answered]
I have a flutter application, and I want to check if the user is logged in, I thought I could use [`window.cookieStore.getAll`](https://api.flutter.dev/flutter/dart-html/CookieStore/getAll.html) ([this](https://developer.mozilla.org/en-US/docs/Web/API/CookieStore/getAll) is the JS function) to check if the cookie exists.

After inspection, however, this function returns empty list even when there is a cookie.

I was hoping to avoid bothering the server for this operation, the cookie does store expiration time; the client-side has all information necessary.

I tested this on `localhost`, maybe this doesn't work on localhost. Anyway, thanks for looking into this!

**Comment by frankie567** [comment, marked as answer]
This is actually expected behavior 😉 By default, cookies are created with the [HTTPOnly flag](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#restrict_access_to_cookies), which explicitly prevents it from being accessed by JavaScript. This is a security best practice since it limits the risk of a malicious user to steal the token through a JS vulnerability.

If you need, you can disable this behavior by setting the `cookie_httponly` argument to `False` on the `CookieTransport`: https://fastapi-users.github.io/fastapi-users/10.1/configuration/authentication/transports/cookie/#configuration

Another solution could be to store basic user information in the local storage and refresh it when needed. This way, you can keep the auth status without reaching the server every time.

**Reply by caniko** [reply]
Oh! That is cool, I'll do the storage strategy :)

Thanks!
